Gemfile

```diff
@@ -12,3 +12,5 @@ when /3/
 else
   gem 'mongoid', version
 end
+
+gem 'mongoid-observers' if version.gsub('~>', '').to_f >= 4
```

Comment:
Still too much magic. Just include it above for the versions you want and that is known to work, and lock at a specific version of `mongoid-overservers` for 4 and 5. Since there's no Gemfile.lock, when there's a new mongoid-observers version released we'd be chasing a moving target otherwise.